Add some explanation to application_services.md (#7091)
matrix-org/synapse: docs/application_services.md

  users:  # List of users we're interested in
    - exclusive: <bool>
      regex: <regex>
      group_id: <group>
    - ...
  aliases: []  # List of aliases we're interested in
  rooms: [] # List of room ids we're interested in
```

> `exclusive`: If enabled registrations of users in that namespace are only allowed for the application service  

Reviewer: ```suggestion
`exclusive`: If enabled, only this application service is allowed to register users in that namespace
```

> `group_id`: All users of this application service are dynamically registered for this group to e.g. use it as flair

Reviewer: ```suggestion
`group_id`: All users of this application service are dynamically joined to this group. This is useful for e.g user organisation or flairs.
```

  rooms: [] # List of room ids we're interested in
```

> `exclusive`: If enabled, only this application service is allowed to register users in that namespace

Reviewer: ```suggestion
`exclusive`: If enabled, only this application service is allowed to register users in its namespace(s).
```
